### ISSUE TITLE
[no-extraneous-import] Error message reports the "@types/blah" package instead of the "blah" package

### ISSUE THREAD

**Issue by cakoose** [import-js/eslint-plugin-import#3208]
Here's a minimal repro: https://github.com/cakoose/no-extraneous-dependencies-reporting-types-package-bug

## Summary of repo contents

index.ts

```
import qs from 'qs';
...
```

package.json (I forget to add a dependency on "qs", but it gets included indirectly via "express")

```
    "dependencies": {
        "express": "^4.21.2"
    },
    "devDependencies": {
        "@types/qs": "^6.9.7",
        ...
    }
```

eslint.config.mjs:

```
    rules: {
        'import/no-extraneous-dependencies': ['error', {
            devDependencies: false,
            peerDependencies: false,
        }],
    },
```

## To reproduce

npm install
npm run lint

```
 .../index.ts
   1:1  error  '@types/qs' should be listed in the project's dependencies, not devDependencies
 import/no-extraneous-dependencies
```

Problem: It's reporting "@types/qs" instead of "qs".

This slight reporting issue really threw me off. I was trying all kinds of weird things to figure out why it was reporting the "@types/qs" package, only to realize that I didn't include "qs" in my dependencies list.

**Comment by ljharb** [comment]
I believe the issue here is something to do with the typescript resolver - @JounQin, is there a solution using eslint-plugin-import for this problem?

**Comment by JounQin** [comment]
I'm unsure why it's an issue caused by the typescript resolver, it resolves `node_modules/@types/qs/index.d.ts` for `qs` specifier intentially and it's correct.

The issue here is that the rule doesn't recognize `qs` specifier itself as dependency but compute `node_modules/@types/qs/index.d.ts` as `@types/qs` instead.
